src/Sunset/EventListener/AddSunsetHeaderListener.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Sunset\EventListener;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+class AddSunsetHeaderListener
```

Comment:
Can you make it `final` please? (sorry I forgot about that before)

Reply:
Seems like a very good idea. Feels way cleaner :) I moved it to Eventlistener/RespondListener, could you please take a look at the test, I'm not 100% certain I did it right.